backend/test/profile.e2e-spec.ts

```diff
@@ -0,0 +1,155 @@
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        DbModule,
+        SharedModule,
+        ConfigModule.forRoot({ isGlobal: true }),
+        PassportModule,
+        rootTypegooseTestModule(),
+        ProfileModule,
+      ],
+    })
+      .overrideProvider(ConfigService)
+      .useClass(MockConfigService)
+      .compile()
+
+    app = await moduleFixture.createNestApplication()
+    app.useGlobalPipes(new ValidationPipe({ whitelist: true }))
+
+    jwtService = app.get<JwtService>(JwtService)
+    connection = await app.get(getConnectionToken())
+    configService = app.get<ConfigService>(ConfigService)
+    userModel = connection.model<User>('User')
+
+    await app.init()
+  })
+
+  afterEach(async () => {
+    await app.close()
+    await closeInMongodConnection()
+    mock.reset()
+  })
+
+  describe('/profile/change-password (POST)', () => {
+    let token
+    beforeEach(async () => {
+      const user = await userModel.create({
+        ...SAMPLE_USER,
+        password: await SAMPLE_USER_PW_HASH(),
+      })
+      token = jwtService.sign(
+        {
+          id: user._id,
+          email: user.email,
+        } as RefreshJWTPayload,
+        { secret: configService.get('JWT_SECRET') },
+      )
+    })
+    describe('Positive Tests', () => {
+      it('should set new password', async () => {
+        // ACT
+        const res = await request(app.getHttpServer())
+          .post('/profile/change-password')
+          .send({
+            oldPassword: SAMPLE_USER.password,
+            password: 'newPassword',
+          })
+          .set({
+            Authorization: `Bearer ${token}`,
+          })
+        // ASSERT
+        expect(res.statusCode).toEqual(HttpStatus.OK)
+        const updatedUser = await userModel.findOne({
+          email: SAMPLE_USER.email,
+        })
+        expect(
+          comparePassword(SAMPLE_USER.password, updatedUser.password),
+        ).toEqual(false)
+        expect(comparePassword('newPassword', updatedUser.password)).toEqual(
+          true,
+        )
+      })
+    })
+
+    describe('Negative Tests', () => {
+      it('should fail for invalid token', async () => {
+        // ACT
+        const res = await request(app.getHttpServer())
+          .post('/profile/change-password')
+          .send({
+            oldPassword: SAMPLE_USER.password,
+            password: 'newPassword',
+          })
+          .set({
+            Authorization: `Bearer ${token}a`,
+          })
+        // ASSERT
+        expect(res.statusCode).toEqual(HttpStatus.UNAUTHORIZED)
+      })
+
+      it('should fail with false password', async () => {
+        // ACT
+        const res = await request(app.getHttpServer())
+          .post('/profile/change-password')
+          .send({
+            oldPassword: SAMPLE_USER.password + 'a',
+            password: 'newPassword',
+          })
+          .set({
+            Authorization: `Bearer ${token}a`,
```

Comment:
```suggestion
            Authorization: `Bearer ${token}`,
```

it would already fail with false token here wouldn't it